### PR TITLE
clusterloader2: Bump Kubernetes dependency to v1.28.12

### PR DESCRIPTION
**Pull request by mboersma**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

This updates the old vendoring of k8s.io in clusterloader2 to a supported Kubernetes version.

#2795 was already open, and I wanted to investigate the remaining issues. I ended up going minor-by-minor release to figure out (mostly) what changes to make, although I still have a `// TODO` in there.

#### Which issue(s) this PR fixes:

Refs #2795

#### Special notes for your reviewer:

@wojtek-t please copy any relevant changes from this PR to yours! I'll be happy to review it.

### CODE REVIEW

**Comment by mboersma** [comment]
/hold
/retitle [WIP] clusterloader2: Bump Kubernetes dependency to v1.28.12

I don't intend to merge this, just want to see how CI treats it and help #2795 get over the finish line.

File: clusterloader2/pkg/framework/config/client_config.go [hunk 1]
```diff
@@ -107,7 +107,7 @@ func transportHack(config *restclient.Config) error {
 		}).DialContext,
 	})
 	config.WrapTransport = transportConfig.WrapTransport
-	config.Dial = transportConfig.Dial
+	// config.Dial = transportConfig.Dial // TODO: What replaces this?
```
Comment (mboersma) [inline clusterloader2/pkg/framework/config/client_config.go:110]: I wasn't sure how to handle this new struct mismatch.

Comment (wojtek-t) [reply clusterloader2/pkg/framework/config/client_config.go:110]: https://github.com/kubernetes/kubernetes/commit/3313a70d5bcc40a39f99f482c18effc9de6072ba shows what to do here

**Review by wojtek-t — COMMENTED** [review]
Thanks a lot for opening this - let's proceed with your PR - I will review it and close mine.

File: clusterloader2/pkg/measurement/common/metrics/metrics_grabber.go [hunk 2]
```diff
@@ -195,7 +195,7 @@ func (g *Grabber) GrabFromControllerManager() (ControllerManagerMetrics, error)
 	var err error
 	podName := fmt.Sprintf("%v-%v", "kube-controller-manager", g.masterName)
 	g.waitForControllerManagerReadyOnce.Do(func() {
-		if readyErr := e2epod.WaitTimeoutForPodReadyInNamespace(g.client, podName, metav1.NamespaceSystem, framework.PodStartTimeout); readyErr != nil {
+		if readyErr := e2epod.WaitTimeoutForPodReadyInNamespace(context.TODO(), g.client, podName, metav1.NamespaceSystem, framework.PodStartTimeout); readyErr != nil {
```
Comment (wojtek-t) [inline clusterloader2/pkg/measurement/common/metrics/metrics_grabber.go:198]: nit: context.Background()

File: clusterloader2/pkg/measurement/util/informer/informer.go [hunk 3]
```diff
@@ -61,7 +61,7 @@ func NewDynamicInformer(
 func addEventHandler(i cache.SharedInformer,
 	handleObj func(interface{}, interface{}),
 ) {
-	i.AddEventHandler(cache.ResourceEventHandlerFuncs{
+	_, _ = i.AddEventHandler(cache.ResourceEventHandlerFuncs{
```
Comment (wojtek-t) [inline clusterloader2/pkg/measurement/util/informer/informer.go:64]: let's not ignore error - let's at least log an error in such case

File: clusterloader2/pkg/provider/autopilot.go [hunk 4]
```diff
@@ -60,7 +62,7 @@ func (p *AutopilotProvider) GetConfig() Config {
 
 func (p *AutopilotProvider) RunSSHCommand(cmd, host string) (string, string, int, error) {
 	// gce provider takes ssh key from GCE_SSH_KEY.
-	r, err := sshutil.SSH(cmd, host, "gce")
+	r, err := sshutil.SSH(context.TODO(), cmd, host, "gce")
```
Comment (wojtek-t) [inline clusterloader2/pkg/provider/autopilot.go:65]: context.Background()

same for other files below

**Review by wojtek-t — COMMENTED** [review]
This is great - I just added couple minor comments

**Comment by mboersma** [comment]
/retitle clusterloader2: Bump Kubernetes dependency to v1.28.12

I updated based on review comments and squashed the commits. Let me know if there's anything else to change.

**Pull request merged**